lib/libspl/asm-generic/atomic.c

```diff
@@ -415,7 +415,7 @@ atomic_clear_long_excl(volatile ulong_t *target, uint_t value)
 
 	VERIFY3S(pthread_mutex_lock(&atomic_lock), ==, 0);
 	bit = (1UL << value);
-	if ((*target & bit) != 0) {
+	if ((*target & bit) != 1) {
```

Comment:
It wouldn't be 1 if `value` is anything but 0 though.
```suggestion
	if ((*target & bit) == 0) {
```